### DISCUSSION TITLE
How to use connection re-usage

### DISCUSSION THREAD

**Discussion by HugoT852** [smartsdk/ngsi-timeseries-api#645] [Q&A] [answered]
Hi,

I was wondering what you need to do to implement the connection re-usage describe here : https://quantumleap.readthedocs.io/en/latest/admin/benchmarks/

Also, is there a redis password Environment Variables?

Thanks in advance.

@agaldemas so he can follow

**Comment by c0c0n3** [comment, marked as answer]
Hi @HugoT852!

Thanks for your questions.

_> I was wondering what you need to do to implement the connection re-usage_

Nothing special, it's built in:

- https://github.com/orchestracities/ngsi-timeseries-api/blob/master/src/utils/connection_manager.py

Needs some improvements, e.g. concurrency, but it should already give you the benefits claimed in the benchmarks...hopefully!

_> Also, is there a redis password Environment Variables?_

Not at the moment. It shouldn't be too hard to implement actually, we welcome PRs :-)
